src/librustc_mir/transform/simplify_cfg.rs

```diff
@@ -126,7 +105,6 @@ impl<'tcx> MirPass<'tcx> for SimplifyCfg {
         while changed {
             changed = self.simplify_branches(mir);
             changed |= self.remove_goto_chains(mir);
-            self.remove_dead_blocks(mir);
```

Comment:
Huh. I'm not sure if it is better to compress the blocks twice for each round. Seems like it might be slower overall?